### ISSUE TITLE
Add training loops

### ISSUE THREAD

**Issue by Fb1234566** [Fb1234566/EdgeMLP#13]
### Training Implementation
- [ ] Implement `void train(X, y, epochs, learning_rate)`
- [ ] Loop through epochs
- [ ] Loop through training samples
- [ ] Forward pass for each sample
- [ ] Backward pass for each sample
- [ ] Update weights after each sample (or batch)
- [ ] Track loss over epochs
- [ ] Print progress every N epochs
